clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreBufferService.java

```diff
@@ -491,9 +506,16 @@ public int hashCode() {
     public boolean equals(Object o) {
       return super.equals(o);
     }
+
+    @Override
+    protected int getBaseClassOverhead() {
+      return PARTIAL_CLASS_OVERHEAD;
+    }
   }
 
   private static class LeaderQueueNode extends QueueNode {
+    private static final int SHALLOW_CLASS_OVERHEAD = ClassSizeEstimator.getClassOverhead(QueueNode.class);
```

Comment:
Why not `LeaderQueueNode`?

Reply:
Good catch! And that is one of the cases I didn't write a unit test for... I think I need to write one now to have a clean conscience 😅

Reply:
I added a unit test for this and it does catch the issue, then I fixed it. Thanks again for catching this!